src/model/fitmodel.py

```diff
@@ -0,0 +1,194 @@
+import pathlib as pl
+from sklearn.metrics import roc_auc_score, roc_curve
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+class AsNumpy:
+    def __init__(self, df):
+        self.df = df
+
+    def train_test_split(self, target_col: str = 'churn', id_col: str = 'customerid'):
+        """
+        This function will split the df into train and test data for model fitting and evalution
+        :param target_col: target col name
+        :param id_col: id col name
+        :return: np.arrays for train_x, train_y, test_x, test_y and list of col names of the data used in modeling
+        """
+
+        # splitting train and test data
+        train, test = train_test_split(self.df, test_size=1 / 4, random_state=42)
+
+        # separating dependent and independent variables
+        cols = [i for i in self.df.columns if i not in id_col + target_col]
+        train_x = train[cols].values
+        train_y = train[target_col].values
+        test_x = test[cols].values
+        test_y = test[target_col].values
+
+        print('train, test data split done!')
+        return train_x, train_y, test_x, test_y, cols
+
+
+class Model:
+    def __init__(self, train_x, train_y, test_x, test_y, prediction_x, cols, algorithm, feature_imp_col):
+        self.train_x = train_x
+        self.train_y = train_y
+        self.test_x = test_x
+        self.test_y = test_y
+        self.cols = cols
+        self.algorithm = algorithm
+        self.feature_imp_col = feature_imp_col
+        self.prediction_x = prediction_x
+
+    def churn_prediction(self):
+        """
+        This function uses the selected algorithm to fit the model
+        :param algorithm: selected algorithm
+        :param train_x: np.array of training x
+        :param test_x: np.array of testing x
+        :param train_y: np.array of training y
+        :param test_y: np.array oftesting y
+        :param cols: list of cols of the df used for model fitting
+        :param feature_imp: str indicating the feature importance variable (coeff. for logistic regression; feature
+        importance scores for tree-based models)
+        :return: a set of model evaluation metrics
+        """
+
+        # train model
+        model = self.algorithm.fit(self.train_x, self.train_y)
+        predictions = model.predict(self.test_x)
+        probabilities = model.predict_proba(self.test_x)
+        print("model fitting done!")
+        return model, predictions, probabilities
+
+    def evaluation(self, predictions, probabilities):
+        fpr, tpr, thresholds = roc_curve(self.test_y, probabilities[:, 1])
+        auc = roc_auc_score(self.test_y, predictions)
+
+        evl = {}
+        evl["accuracy"] = accuracy_score(self.test_y, predictions)
+        evl["classification report"] = classification_report(self.test_y, predictions)
+        evl["confusion materix"] = confusion_matrix(self.test_y, predictions)
+        print("evalution done!")
+        return evl, fpr, tpr, auc
```

Comment:
optional, probably better to output a dictionary with metric name.

Reply:
Do you mean sth like {'train_x': train_x, 'train_y': train_y, ...}? Why is dic better than np.arrays? 
One output vs. Four?